src/_pytest/python_api.py

```diff
@@ -265,19 +265,20 @@ def _repr_compare(self, other_side: Mapping[object, float]) -> List[str]:
             approx_side_as_map.items(), other_side.values()
         ):
             if approx_value != other_value:
-                max_abs_diff = max(
-                    max_abs_diff, abs(approx_value.expected - other_value)
-                )
-                if approx_value.expected == 0.0:
-                    max_rel_diff = math.inf
-                else:
-                    max_rel_diff = max(
-                        max_rel_diff,
-                        abs(
-                            (approx_value.expected - other_value)
-                            / approx_value.expected
-                        ),
+                if not any((approx_value.expected is None, other_value is None)):
```

Comment:
Suggestion: I personally prefer to be explicit, using `any` here makes the code needlessly more complex than it needs to be.

```suggestion
                if approx_value.expected is not None and other_value is not None:
```

Reply:
Works for me, seeing both I do agree it's more clear than negating the ``any``. Thanks!